lib/ansible/modules/remote_management/manageiq/manageiq_user.py

```diff
@@ -0,0 +1,239 @@
+
+ANSIBLE_METADATA = {'metadata_version': '1.0',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+
+DOCUMENTATION = '''
+
+module: manageiq_user
+
+short_description: Management of users in ManageIQ
+extends_documentation_fragment: manageiq
+version_added: '2.4'
+author: Daniel Korn (@dkorn)
+description:
+  - The manageiq_user module supports adding, updating and deleting users in ManageIQ.
+
+options:
+  action:
+    description:
+      - Specifies the action to take.
+    required: False
+    choices: ['create', 'delete', 'edit']
+    default: 'create'
+  userid:
+    description:
+      - The unique userid in manageiq, often mentioned as username.
+    required: true
+  name:
+    description:
+      - The users' full name.
+    required: false
+    default: null
+  password:
+    description:
+      - The users' password.
+    required: false
+    default: null
+  group:
+    description:
+      - The name of the group to which the user belongs.
+    required: false
+    default: null
+  email:
+    description:
+      - The users' E-mail address.
+    required: false
+    default: null
+'''
+
+EXAMPLES = '''
+- name: Create a new user in ManageIQ
+  manageiq_user:
+    action: 'create'
+    userid: 'jdoe'
+    name: 'Jane Doe'
+    password: 'VerySecret'
+    group: 'EvmGroup-user'
+    email: 'jdoe@example.com'
+    miq:
+      url: 'http://example.com:3000'
+      username: 'admin'
+      password: 'smartvm'
+      verify_ssl: False
+'''
+
+RETURN = '''
+'''
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.manageiq import (
+    check_client,
+    ManageIQ,
+)
+
+
+class ManageIQUser(object):
+    """
+        object to execute user management operations in manageiq
+    """
+
+    def __init__(self, manageiq):
+        self.manageiq = manageiq
+        self.api_url = self.manageiq.api_url
+
+    def group_id(self, group):
+        """ Search for group id by group name
+
+        Returns:
+            the group id, or send a module Fail signal if group not found
+        """
+        group_obj = self.manageiq.find_collection_resource_by('groups', description=group)
+        if not group_obj:  # group doesn't exist
+            self.manageiq.module.fail_json(
+                msg="Group {group} does not exist in manageiq".format(group=group))
+
+        return group_obj['id']
+
+    def user(self, userid):
+        """ Search for user object by userid
+
+        Returns:
+            the user, or send a module Fail signal if group not found
+        """
+        user = self.manageiq.find_collection_resource_by('users', userid=userid)
+        if not user:  # user doesn't exist
+            self.manageiq.module.fail_json(
+                msg="User {userid} does not exist in manageiq".format(userid=userid))
+
+        return user
+
+    def delete_user(self, userid):
+        """Deletes a user from manageiq.
```

Comment:
this is not good way to set 'changed', if user does not exist you are erroring out instead of returning 'ok' and `changed=False`